code/obj/item/assembly/misc_assemblies.dm

```diff
@@ -96,8 +96,8 @@ Contains:
 		return
 	if (iswrenchingtool(W) && !(src.status))
 		var/turf/T = src.loc
-		if (ismob(T))
-			T = T.loc
+		if (ismob(T) || istype(T, /obj/item/storage))
+			T = get_turf(src)
```

Comment:
I think I'd just do `var/turf/T = get_turf(src)`, without the if. I can't even come up with a scenario in which you could disassemble the thing without it being in storage on turf or on your person. The same applies to the other changed bits of code.

Sidenote: Disassembling an assembly while you are inside a locker or a vehicle will drop the parts out of that locker / vehicle now. But that's probably fine.

Reply:
Done.